src/js/jsx/search/SearchBar.jsx

```diff
@@ -91,7 +93,8 @@ define(function (require, exports, module) {
             this.setState({
```

Comment:
Above, it worries me a little bit that this modifies the search store at mount time. Can you explain what that `_update` call accomplishes? Also, it seems like this should be a `StoreWatchMixin` so that it could pick up new search providers registered after this component's first mount time.